botocore/useragent.py

```diff
@@ -71,19 +103,34 @@ def sanitize_user_agent_string_component(raw_str, allow_hash):
     )
 
 
+class UserAgentSizeConfig(NamedTuple):
```

Comment:
Should this be `UserAgentComponentSizeConfig`?

Reply:
To provide more context, I'm suggesting this incase we ever decide to add a size config for the entire user agent string and not just a component. We can probably leave the name as is if we think this same config could be reused for that case. If not, renaming might make sense.

Reply:
Addressed in https://github.com/boto/botocore/pull/3389/commits/a743c33783692038adf7f573449eab721e06fab2